conn/network.go

```diff
@@ -34,6 +34,21 @@ import (
 type response struct {
 	result chan *gauge_messages.Message
 	err    chan error
+	timer  *time.Timer
+}
+
+func (r *response) stopTimer() {
+	if r.timer != nil {
+		r.timer.Stop()
+	}
+}
+
+func (r *response) addTimer(timeout time.Duration, messageType string) {
+	if timeout > 0 {
+		r.timer = time.AfterFunc(timeout, func() {
+			r.err <- fmt.Errorf("%s request timed out.", messageType)
```

Comment:
Should perhaps also have the message ID. The older message was showing which message timed out, which is useful when looking at API logs.